proxy/src/fully_qualified_authority.rs

```diff
@@ -212,6 +212,29 @@ mod tests {
         assert_eq!("name.namespace.svc.cluster.local",
                    local("name.namespace.svc.cluster.local.", "namespace"));
 
+        // Irrespective of how other absolute names are resolved, "localhost."
+        // absolute names aren't ever resolved through the destination service,
+        // as prescribed by https://tools.ietf.org/html/rfc6761#section-6.3:
+        //
+        //     The domain "localhost." and any names falling within ".localhost."
+        //     are special in the following ways: [...]
+        //
+        //     Name resolution APIs and libraries SHOULD recognize localhost
+        //     names as special and SHOULD always return the IP loopback address
+        //     for address queries [...] Name resolution APIs SHOULD NOT send
+        //     queries for localhost names to their configured caching DNS server(s).
```

Comment:
This isn't a blocker or anything but it seems to me that it might eventually be nice if some of this description of the tests was moved out of comments and into the test assertions, so that the failure messages would explain _why_ the assertions were made. Of course, the best way to go about this might require some additional work to try and support a more BDD-like testing style in Rust, so it may be out of scope. Just a thought.